File: blog-backend/blog-core/src/main/java/com/zach/blog/service/ArticleService.java
package com.zach.blog.service;

import com.zach.blog.dto.request.UpdateArticleRequest;
import com.zach.blog.dto.request.WriteArticleRequest;
import com.zach.blog.model.ApplicationUser;
import com.zach.blog.model.Article;
import org.springframework.data.domain.Page;
import org.springframework.web.multipart.MultipartFile;

import java.io.IOException;
import java.util.List;

public interface ArticleService {
    List<Article> getHotArticles();

    Page<Article> getArticles(Integer pageNum, Integer pageSize, Long categoryId);

    Article getArticleDetail(Long categoryId);

    void updateViewCount(Long articleId);

    void createArticle(Long userId, WriteArticleRequest writeArticleRequest) throws IOException;

    Page<Article> getArticles(Integer pageNum, Integer pageSize, String title, String summary);

    Article updateArticle(Long id, UpdateArticleRequest updateArticleRequest) throws IOException;

    void deleteArticle(Long id);

    void updateArticleImage(Long articleId, ApplicationUser user, MultipartFile image) throws IOException;
}

File: blog-frontend/blog-admin/src/services/resources/article-service.ts
import { resourceAxios } from "../../utils/axios-utils"

export const articleService = {
    createArticle: async (data: WriteArticleRequest) => {
        const formData = new FormData()
        Object.entries(data).forEach(([key, value]) => {
            formData.append(key, value)
        })

        await resourceAxios.post<void, void>('/articles', formData, {
            headers: {
                'Content-Type': 'multipart/form-data'
            }
        })
    },

    getArticles: async ({ pageSize = 5, pageNum = 0, title, summary }: GetArticles) => {
        const result = await resourceAxios.get<void, PageRespsone<Article>>('/articles', {
            params: {
                pageSize,
                pageNum,
                ...(title !== '' ? { title } : {}),
                ...(summary !== '' ? { summary } : {})
            }
        })
        return result
    },

    getArticleDetails: async (id: number) => {
        const result = await resourceAxios.get<void, ArticleDetails>(`/articles/${id}`)
        return result
    },

    updateArticle: async ({ id, article }: UpdateArticleRequest) => {
        const formData = new FormData()
        Object.entries(article).forEach(([key, value]) => {
            formData.append(key, value)
        })

        const result = await resourceAxios.put<void, Article>(`/articles/${id}`, formData, {
            headers: {
                'Content-Type': 'multipart/form-data'
            }
        })
        return result
    },

    deleteArticle: async (id: number) => {
        const result = await resourceAxios.delete<void, void>(`/articles/${id}`)
        return result
    }

}